agent/lib/kontena/workers/service_pod_worker.rb

```diff
@@ -195,10 +195,16 @@ def container_outdated?(service_container)
     # @param [Docker::Container] service_container
     # @return [Boolean]
     def image_outdated?(service_container)
-      image = Docker::Image.get(service_pod.image_name) rescue nil
-      return true unless image
-      return true if image.id != service_container.info['Image']
-
+      begin
+        image = Docker::Image.get(service_pod.image_name)
+        if image.id != service_container.info['Image']
+          info "containers image(#{service_container.info['Image']} does not match images id(#{image.id})}"
+          return true
+        end
+      rescue Docker::Error::NotFoundError
+        info "failed to get image: #{service_pod.image_name}"
```

Comment:
Not directly on this code path, but there seems to be cases when docker gives `No such image` although image listing shows them in https://github.com/kontena/kontena/issues/1979